File: 2017/02.fs
namespace _2017

open System

module _02 =
    let Data = (Utils.GetInputData 2).Split('\n', StringSplitOptions.RemoveEmptyEntries)

    let parseLine(line: string): int[] =
        line.Split('\t', StringSplitOptions.RemoveEmptyEntries) |> Array.map int

    let findEvenDivision(numbers: int[]) =
        numbers
        |> Array.collect(fun x -> numbers |> Array.map(fun y -> if x <> y && x % y = 0 then Some (x / y) else None))
        |> Array.choose id
        |> Array.head

    let solvePartOne() =
        Data |> Array.fold(fun sum line ->
            let numbers = parseLine line |> Array.sort
            sum + (numbers[numbers.Length - 1] - numbers[0])
        ) 0

    let solvePartTwo() =
        Data |> Array.fold(fun acc line ->
            let numbers = parseLine line
            acc + (findEvenDivision numbers)
        ) 0

    let Run() =
        printfn $"Part One: {solvePartOne()}"
        printfn $"Part Two: {solvePartTwo()}"

File: 2018/02.js
import {getInputData} from './utils.js';

const data = (await getInputData(2)).trim().split('\n');

function solvePartOne() {
	let twoCounts = 0;
	let threeCounts = 0;
	for (const id of data) {
		const charMap = {};
		for (const char of id) charMap[char] = (charMap[char] || 0) + 1;
		if (Object.values(charMap).some(x => x === 2)) twoCounts++;
		if (Object.values(charMap).some(x => x === 3)) threeCounts++;
	}
	return twoCounts * threeCounts;
}

function solvePartTwo() {
	for (let i = 0; i < data.length; i++) for (let j = i + 1; j < data.length; j++) {
		const id1 = data[i];
		const id2 = data[j];
		let differingCount = 0;
		let commonChars = '';
		for (let k = 0; k < id1.length; k++) {
			if (id1[k] === id2[k]) commonChars += id1[k];
			else differingCount++;
			if (differingCount > 1) break;
		}
		if (differingCount === 1) return commonChars;
	}
}

console.log(`Part One: ${solvePartOne()}`);
console.log(`Part Two: ${solvePartTwo()}`);